packages/pegasus/src/pegasus.js

```diff
@@ -1,266 +1,47 @@
-      const packet = localAmountToPacket(amount, depositAddress);
-
-      // Retain the payment.  We must not proceed on failure.
-      retain(zcfSeat, { Transfer: amount });
-
-      // The payment is already escrowed, and proposed to retain, so try sending.
-      return sendTransferPacket(connection, packet).then(
-        _ => zcfSeat.exit(),
-        reason => {
-          // Return the payment to the seat, if possible.
-          redeem(zcfSeat, { Transfer: amount });
-          throw reason;
-        },
-      );
-    };
-
-    // Reflect any error back to the seat.
-    return tryToSend().catch(reason => {
-      zcfSeat.fail(reason);
-    });
-  };
-
-  /** @type {Receiver} */
-  const receive = async packet => {
-    // Look up the deposit facet for this board address, if there is one.
-    const depositAddress = packet.receiver;
-    /** @type {DepositFacet} */
-    const depositFacet = await E(board)
-      .getValue(depositAddress)
-      .catch(_ => E(namesByAddress).lookup(depositAddress, 'depositFacet'));
-    const localAmount = packetToLocalAmount(packet);
-
-    const { userSeat, zcfSeat } = zcf.makeEmptySeatKit();
-
-    // Redeem the backing payment.
-    try {
-      redeem(zcfSeat, { Transfer: localAmount });
-      zcfSeat.exit();
-    } catch (e) {
-      zcfSeat.fail(e);
-      throw e;
-    }
-
-    // Once we've gotten to this point, their payment is committed and
-    // won't be refunded on a failed receive.
-    const payout = await E(userSeat).getPayout('Transfer');
-
-    // Send the payout promise to the deposit facet.
-    E(depositFacet)
-      .receive(payout)
-      .catch(_ => {});
-
-    // We don't want to wait for the depositFacet to return, so that
-    // it can't hang up (i.e. DoS) an ordered channel, which relies on
-    // us returning promptly.
-    return undefined;
-  };
-
-  return Far('courier', { send, receive });
-};
-
 /**
  * Make a Pegasus public API.
  *
  * @param {ContractFacet} zcf the Zoe Contract Facet
- * @param {BoardDepositFacet} board where to find depositFacets by boardID
- * @param {NameHub} namesByAddress where to find depositFacets by bech32
+ * @param {ERef<BoardDepositFacet>} board where to find depositFacets by boardID
+ * @param {ERef<NameHub>} namesByAddress where to find depositFacets by bech32
  */
 const makePegasus = (zcf, board, namesByAddress) => {
-  /** @type {NotifierRecord<Peg[]>} */
-  const { notifier, updater } = makeNotifierKit([]);
-
   /**
    * @typedef {Object} LocalDenomState
-   * @property {Store<DenomUri, Courier>} denomUriToCourier
-   * @property {Set<Peg>} pegs
+   * @property {Address} localAddr
+   * @property {Address} remoteAddr
+   * @property {Store<Denom, PromiseRecord<Courier>>} remoteDenomToCourierPK
+   * @property {IterationObserver<Denom>} remoteDenomPublication
+   * @property {Subscription<Denom>} remoteDenomSubscription
    * @property {number} lastDenomNonce
+   * @property {(reason?: any) => void} abort
```

Comment:
I notice `CloseReason` is already defined as `any?` for this purpose.
```suggestion
   * @property {(reason: CloseReason) => void} abort
```